src/main/scala/firrtl/PrimOps.scala

```diff
@@ -305,16 +238,11 @@ object PrimOps extends LazyLogging {
         case (t1, t2) => UnknownType
       }
       case Bits => t1 match {
-        case (_: UIntType | _: SIntType) => UIntType(PLUS(MINUS(c1, c2), IntWidth(1)))
-        case _: FixedType => UIntType(PLUS(MINUS(c1, c2), IntWidth(1)))
-        case _ => UnknownType
-      }
-      case Head => t1 match {
-        case (_: UIntType | _: SIntType | _: FixedType) => UIntType(c1)
+        case (_: UIntType | _: SIntType | _: FixedType) => UIntType(PLUS(MINUS(c1, c2), IntWidth(1)))
         case _ => UnknownType
       }
-      case Tail => t1 match {
-        case (_: UIntType | _: SIntType | _: FixedType) => UIntType(MINUS(w1, c1))
+      case ht @(Head | Tail) => t1 match {
+        case (_: UIntType | _: SIntType | _: FixedType) => if (ht == Head) UIntType(c1) else UIntType(MINUS(w1, c1))
```

Comment:
To me, this reads better unmodified. It seems counterintuitive to match on both ops then handle each op separately.